packages/app/src/elements/field.ts

```diff
@@ -200,6 +228,123 @@ export class FieldElement extends BaseElement {
         }
     }
 
+    /**
+     * Checks against HIBP's API for the number of times the user's password has been detected in a breach.
+     * @private
+     */
+    private async _checkPasswordBreach() {
+        if (!this.value) {
+            this._passwordBreachCount = -1;
+            return;
+        }
+
+        const sha1Hash = crypto.createHash("sha1")
+            .update(this.value)
+            .digest("hex");
+
+        try {
+            const response = await app.getPasswordBreachStatus(sha1Hash);
+            this._passwordBreachCount = response.count;
+        } catch (error) {
+            console.error(error);
+            this._passwordBreachCount = -1;
+        }
+    }
+
+    /**
+     * Get the CSS style class for an input field, based on the strength of the password in this field.
+     * Password strength is computed by checking for uppercase characters, lowercase characters, special
+     * characters, and numbers.
+     * @private
+     */
+    private _getPasswordStrengthClass(): string {
+        this._missingList = [];
+        if (!this.value) {
+            return "value-input";
+        }
+
+        let strength = 0;
+
+        // check password length
+        if (this.value.length > 15) {
+            strength += 4;
+        } else if (this.value.length > 10) {
+            strength += 2;
+        } else if (this.value.length > 5) {
+            strength += 1;
+        }
+
+        // check for uppercase
+        const containsUppercase = (/[A-Z]/g).test(this.value);
+        if (containsUppercase) {
+            strength += 2;
+        } else {
+            this._missingList.push("uppercase letter");
+        }
+
+        // check for lowercase
+        const containsLowercase = (/[a-z]/g).test(this.value);
+        if (containsLowercase) {
+            strength += 2;
+        } else {
+            this._missingList.push("lowercase letter");
+        }
+
+        // check for number
+        const containsNumber = (/\d/g).test(this.value);
+        if (containsNumber) {
+            strength += 2;
+        } else {
+            this._missingList.push("number");
+        }
+
+        // check for special character
+        const containsSpecial = (/[-+_!@#$%^&*.,?;]/g).test(this.value);
+        if (containsSpecial) {
+            strength += 2;
+        } else {
+            this._missingList.push("special character");
+        }
+
+        if (strength >= 12) {
+            return "value-input value-strong";
+        } else if (strength > 8) {
+            return "value-input value-good";
+        } else if (strength > 6) {
+            return "value-input value-okay";
+        }
+
+        return "value-input value-weak";
+    }
```

Comment:
This is a very simplistic way of "calculating" password strength and while it's better than nothing, it's certainly not an accurate or reliable way to measure password entropy. Besides, we already have a [much better way](https://github.com/padloc/padloc/blob/master/packages/app/src/lib/util.ts#L25) to do this, which based on the [zxcvbn](https://github.com/dropbox/zxcvbn) library.